lib/iris/fileformats/pp.py

```diff
@@ -1513,9 +1545,34 @@ def t2(self, dt):
         self.lbhrd = dt.hour
         self.lbmind = dt.minute
         self.lbdayd = int(dt.strftime("%j"))
+        if self.lbyrd == 0:
+            if self.lbmond == 0 and self.lbdatd == 0:
+                self.lbmond = self.lbdatd = 1
+            elif self.lbmond == 0 or self.lbdatd == 0:
+                emsg = (
+                    f"Invalid {self.__class__.__name__} t2 date {dt} provided"
+                )
+                raise ValueError(emsg)
         if hasattr(self, "_t2"):
             delattr(self, "_t2")
 
+    @property
+    def t2_zero(self):
+        """
+        Initialise the associated ``t2`` field properties to zero.
+
+        It is not possible to do this though the setter property as true zero
+        dates are not valid :func:`datetime.datetime` or :func`cftime.datetime`
+        dates.
+
+        """
+        self.lbyrd = 0
+        self.lbmond = 0
+        self.lbdatd = 0
+        self.lbhrd = 0
+        self.lbmind = 0
+        self.lbdayd = 0
```

Comment:
We only have the use case of wanting to zero `t2` date field, not the `t1` date field - so I've not included that behaviour.